server/app/src/main/java/com/google/heartrate/wearos/app/gatt/heartrate/characteristics/HeartRateMeasurementCharacteristic.java

```diff
@@ -0,0 +1,269 @@
+
+    /** Heart Rate Measurement characteristic UUID. */
+    private static final UUID HEART_RATE_MEASUREMENT_UUID = UUID
+            .fromString("00002a37-0000-1000-8000-00805f9b34fb");
+
+    /** Indicates that the data format of the Heart Rate Measurement value field is in a format of UINT8.  */
+    private static final byte HEART_RATE_MEASUREMENT_UINT8_FLAG = 0b00000000;
+
+    /** Indicates that the data format of the Heart Rate Measurement value field is in a format of UINT16. */
+    private static final byte HEART_RATE_MEASUREMENT_UINT16_FLAG = 0b00000001;
+
+    /** Indicates that the Energy Expended field is present in the Heart Rate Measurement characteristic. */
+    private static final byte EXPENDED_ENERGY_FLAG = 0b00001000;
+
+
+    /** Offset at which the Heart Rate Measurement characteristic flags are stored. */
+    private static final int FLAGS_OFFSET = 0;
+
+    /** Offset at which Heart Rate Measurement value is stored. */
+    private static final int HEART_RATE_MEASUREMENT_OFFSET = 1;
+
+    /** Offset at which Expended Energy value is Heart Rate Measurement value field is in a format of UINT8. */
+    private static final int EXPENDED_ENERGY_BASE_OFFSET = 2;
+
+    /** Offset at which Expended Energy value is Heart Rate Measurement value field is in a format of UINT16. */
+    private static final int EXPENDED_ENERGY_SHIFTED_OFFSET = 3;
+
+    /**
+     * Create {@link GattCharacteristic} for Heart Rate Measurement characteristic.
+     * <br>Configure {@link GattCharacteristic} with Heart Rate Measurement characteristic UUID,
+     * property for notification, no read/write permissions and Client Characteristic Configuration Descriptor.
+     */
+    public HeartRateMeasurementCharacteristic() {
+        super(HEART_RATE_MEASUREMENT_UUID,
+                BluetoothGattCharacteristic.PROPERTY_NOTIFY,
+                BluetoothGattCharacteristic.PROPERTY_NOTIFY,
+                new GattDescriptor[] {new ClientCharacteristicConfigurationDescriptor()});
+    }
+
+    /**
+     * Get heart rate measurement value.
+     *
+     * @return heart rate measurement value
+     * @throws GattException in case of wrong format or offset for heart rate measurement value
+     */
+    public int getHeartRateMeasurementValue() throws GattException {
+        Log.i(TAG, "Get heart rate measurement value");
+
+        int format = isUInt16HeartRateFormat() ?
+                BluetoothGattCharacteristic.FORMAT_UINT16 :
+                BluetoothGattCharacteristic.FORMAT_UINT8;
+
+        int heartRateMeasurement = getIntValue(format, HEART_RATE_MEASUREMENT_OFFSET);
+
+        Log.v(TAG, String.format("Heart rate format=%d value=%d", format, heartRateMeasurement));
+
+        return heartRateMeasurement;
+    }
+
+    /**
+     * Get expended energy value.
+     *
+     * @return expended energy value
+     * @throws GattException if expended energy is not presented or in case of wrong format or offset for expended energy value
```

Comment:
typo: present

Reply:
Fixed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/1/commits/f81955e6ee20af370c211fd96123abd538b599e1l)